docs: fixed a bunch of things in the docs (#2653)
operator-framework/operator-sdk: doc/ansible/user-guide.md

  group: baz.example.com
  kind: Baz
  playbook: baz.yml

>   reconcilePeriod: 10s

Reviewer: ```suggestion
  reconcilePeriod: 0s
```

```yaml
---
- name: start memcached
  k8s:

>     state: present

Reviewer: You replaced it because you face: 

 ERROR! couldn't resolve module/action 'community.kubernetes.k8s'. This often indicates a misspelling, missing collection, or incorrect module path.                                                                                                      │
│ The error appears to be in '/opt/ansible/roles/memcached/tasks/main.yml': line 2, column 3, but may                                                                                                                                                      │
│ be elsewhere in the file depending on the exact syntax problem.                                                                                                                                                                                          │
│ The offending line appears to be:                                                                                                                                                                                                                        │
│ ---                                                                                                                                                                                                                                                      │
│ - name: start memcached                                                                                                                                                                                                                                  │
│   ^ here

However, it is the master branch and not the latest tag release. Did you check it with the master branch?
See that to use the master branch your Dockerfile should be as follows:

```
FROM quay.io/operator-framework/ansible-operator:master

COPY watches.yaml ${HOME}/watches.yaml
COPY roles/ ${HOME}/roles
```

And you need to update your project with all break changes.  See; https://github.com/operator-framework/operator-sdk/blob/master/doc/migration/version-upgrade-guide.md#breaking-changes-2

Reviewer: Note that it was changed for the changes made in the master and the doc of the latest release is as you sugessted here; https://github.com/operator-framework/operator-sdk/blob/v0.15.x/doc/ansible/user-guide.md

Author: Ah, I have used the version installed through homebrew which is also the latest version.

```
~ $ operator-sdk version
operator-sdk version: "v0.15.2", commit: "ffaf278993c8fcb00c6f527c9f20091eb8dd3352", go version: "go1.13.8 darwin/amd64"
~ $
```

Kubernetes deployment manifests are generated in `deploy/operator.yaml`. The
deployment image in this file needs to be modified from the placeholder
`REPLACE_IMAGE` to the previous built image.

> Also, we need to update the `imagePullPolicy` to `Always` (makes it easier for testing).

Reviewer: ```suggestion
Also, we recommend keeping the `imagePullPolicy` as `Always` because it makes easier for testing.
```

To do the above run:
```

> $ sed -i 's|image:.*|image: quay.io/example/memcached-operator:v0.0.1|g' deploy/operator.yaml

Reviewer: Why you change this one?

Reviewer: Thee current sed command works perfect.

Reviewer: So, could you please revert these ones?

Author: Hey @camilamacedo86 ,

Thanks for looking into it. Please see the below explanation and let me know if I have missed anything. I had to make these changes to make things work in `v0.15.2`

The `REPLACE_IMAGE` is an Ansible variable. If we just replace it with a value, Ansible will throw an error as `quay.io/example/memcached-operator:v0.0.1` is not a variable.
```
~/.../memcached-operator (ansible) $ grep REPLACE_IMAGE deploy/operator.yaml 
          image: "{{ REPLACE_IMAGE }}"
          image: "{{ REPLACE_IMAGE }}"
~/.../memcached-operator (ansible) $ sed -i 's|REPLACE_IMAGE|quay.io/example/memcached-operator:v0.0.1|g' deploy/operator.yaml
~/.../memcached-operator (ansible) $ grep "image:" deploy/operator.yaml 
          image: "{{ quay.io/example/memcached-operator:v0.0.1 }}"
          image: "{{ quay.io/example/memcached-operator:v0.0.1 }}"
~/.../memcached-operator (ansible) $
```

And we need to have a value against `imagePullPolicy`, otherwise Kubernetes will throw this error.
```bash
~/.../memcached-operator (ansible) $ kubectl apply -f deploy/operator.yaml
The Deployment "memcached-operator" is invalid: 
* spec.template.spec.containers[0].imagePullPolicy: Unsupported value: "{{ pull_policy|default('Always') }}": supported values: "Always", "IfNotPresent", "Never"
* spec.template.spec.containers[1].imagePullPolicy: Unsupported value: "{{ pull_policy|default('Always') }}": supported values: "Always", "IfNotPresent", "Never"
~/.../memcached-operator (ansible) $
```

> $ sed -i 's|imagePullPolicy:.*|imagePullPolicy: Always|g' deploy/operator.yaml

Reviewer: Not required be added. See above that default value scaffolded will be Always.

Author: @camilamacedo86  - I think we still need to do the same, otherwise, how will these values be fixed before running this command `$ kubectl create -f deploy/operator.yaml` as suggested in the user [guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/user-guide.md#1-run-as-a-pod-inside-a-kubernetes-cluster).

I am new to this project, have I missed any steps?

Author: Ah, I see what you mean now? User guide with [v0.15.X](https://github.com/operator-framework/operator-sdk/blob/v0.15.x/doc/ansible/user-guide.md) is slightly different from that of the [master](https://github.com/operator-framework/operator-sdk/blob/master/doc/ansible/user-guide.md).

Let me try it out quickly, will update the PR accordingly.

Author: Ah, those changes are already mentioned in the [v0.15.X](https://github.com/operator-framework/operator-sdk/blob/v0.15.x/doc/ansible/user-guide.md). The user guide in `master` branch is different.
I am updating my PR and will probably raise a PR against `v0.15.x` branch.

```

**Note**
If you are performing these steps on OSX, use the following `sed` commands instead:
```
$ sed -i "" 's|image:.*|image: quay.io/example/memcached-operator:v0.0.1|g' deploy/operator.yaml

> $ sed -i "" 's|imagePullPolicy:.*|imagePullPolicy: Always|g' deploy/operator.yaml

Reviewer: Ditto.
